point_pattern.py

```diff
@@ -119,7 +142,24 @@ def average_nearest_neighbor_distance(points):
      Measure of Spatial Relationships in Populations. Ecology. 35(4)
      p. 445-453.
     """
-    mean_d = 0
+
+    min_dist_sum = 0
+    for coord_n in points:
+        first = True
+        for  coord_m in points:
+            if coord_n == coord_m:
+                continue
```

Comment:
Check out `enumerate`.  The logic you have here works wonderfully, until the points are coincident.  If we knew the positional index for the outer loop point, we could skip that index for the inner loop point.